fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/settings/SettingsViewModel.java

```diff
@@ -0,0 +1,68 @@
+package com.github.dedis.popstellar.ui.settings;
+
+import android.app.Application;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.AndroidViewModel;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import com.github.dedis.popstellar.SingleEvent;
+
+import io.reactivex.disposables.CompositeDisposable;
+
+public class SettingsViewModel extends AndroidViewModel {
+
+  public static final String TAG = SettingsViewModel.class.getSimpleName();
+
+  /*
+   * LiveData objects for capturing events like button clicks
+   */
+  private final MutableLiveData<SingleEvent<Boolean>> mApplyChangesEvent = new MutableLiveData<>();
+
+  /*
+   * LiveData objects that represent the state in a fragment
+   */
+  private final MutableLiveData<String> mServerUrl = new MutableLiveData<>();
+  private final MutableLiveData<String> mTempServerUrl = new MutableLiveData<>();
```

Comment:
It could be a little bit confusing which one is what only based on name, without comment,
Is it normal that the one used directly is mServerUrl and the one that keep the last url is mTempServerUrl ?

Reply:
mTempServerUrl stores the server URL at the moment we launch settings, that's why we're checking with that one. mServerUrl is updated when we're writing the server URL in the entry box meaning it keeps the last one.

Reply:
but I changed the name for mCheckServerUrl which is les confusing and added comment to explain its purpose